### PR TITLE
HPCC-30140 Initial test cases for roxie https soapcall

### PR DESCRIPTION
**Pull request by ghalliday**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by afishbeck** [comment]
@ghalliday I think this is a fine starting point.  I'm not sure you'll like some of the things we do to put a structured ESDL interface in front.  It will give you a chance to put your 2 cents in.

We will probably separate out a url input/stored for each of the soapcalls.  Mostly to emulate how things are done in production.

File: testing/regress/ecl/setup/soaptest_getdocumentcount.ecl [hunk 1]
```diff
@@ -0,0 +1,7 @@
+//nothor
```
Comment (afishbeck) [inline testing/regress/ecl/setup/soaptest_getdocumentcount.ecl:1]: I think even in the regression suite we usually put a comment block at the top of every ecl file.

File: testing/regress/ecl/common/SoapTextTest.ecl [hunk 2]
```diff
@@ -0,0 +1,132 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2012 HPCC Systems®.
```
Comment (afishbeck) [inline testing/regress/ecl/common/SoapTextTest.ecl:3]: Copy and paste of an old date.

File: testing/regress/ecl/soaptext1.ecl [hunk 3]
```diff
@@ -0,0 +1,33 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2012 HPCC Systems®.
```
Comment (afishbeck) [inline testing/regress/ecl/soaptext1.ecl:3]: Copy and paste of an old date.

**Review by afishbeck — CHANGES_REQUESTED** [review]
Minor comments about copy right dates and file headings.

The url inputs would likely be split into one for each soapcall, and there might be other minor changes, when we put an ESDL interface in front.

**Review by afishbeck — COMMENTED** [review]
@ghalliday A few comments, but the test itself looks fine to me.

It would be nice if we could get the regression test to actually test https and maybe secrets.  A first step could be doing that for kubernetes.

Having the test suite write a secret to disk in bare-metal might not be that hard for a single node system.

Beyond just supporting https soapcalls, in the cloud test repository we can make our queries work properly with scapps and desdl.

File: testing/regress/ecl/common/SoapTextTest.ecl [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+    EXPORT doDocumentCount(string search) := FUNCTION
+        cleanedSearch := Std.Str.ToLowerCase(TRIM(search));
+        searchIndex := files.getSearchIndexVariant(variant);
+        matches := searchIndex(KEYED(kind = Ts.KindType.TextEntry AND word = cleanedSearch)) : onwarning(4523, ignore);
+        bydocs := TABLE(matches, { cnt := COUNT(GROUP), doc }, doc);
+        numDocs := COUNT(byDocs);
+        RETURN numDocs;
+    END;
+
+    EXPORT countServiceRequest := RECORD
```
[the remark this answers is not in the document]
Comment (ghalliday) [reply testing/regress/ecl/common/SoapTextTest.ecl:44]: Yes.  I will switch so it uses them, but I will save switching to the ESDL style as a separate PR.

**Review by afishbeck — APPROVED** [review]

**Pull request merged**